metron-platform/metron-integration-test/src/main/java/org/apache/metron/integration/components/YarnComponent.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.metron.integration.components;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileContext;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.util.JarFinder;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import org.apache.hadoop.yarn.server.MiniYARNCluster;
+import org.apache.hadoop.yarn.server.nodemanager.NodeManager;
+import org.apache.hadoop.yarn.server.resourcemanager.scheduler.capacity.CapacityScheduler;
+import org.apache.metron.integration.InMemoryComponent;
+import org.apache.metron.integration.UnableToStartException;
+
+import java.io.*;
+import java.net.URL;
+
+public class YarnComponent implements InMemoryComponent{
+    protected MiniYARNCluster yarnCluster = null;
+    protected YarnConfiguration conf = null;
+    private static final int NUM_NMS = 1;
+    private String testName = null;
+    protected String appmasterJar = null;
+
+    public YarnComponent withApplicationMasterClass(Class clazz){
+       appmasterJar = JarFinder.getJar(clazz);
+        return this;
+    }
+
+    public YarnComponent withTestName(String name){
+        this.testName = name;
+        return this;
+    }
+
+    public String getAppMasterJar(){
+        return appmasterJar;
+    }
+
+    public YarnConfiguration getConfig(){
+        return conf;
+    }
+
```

Comment:
One question here is to expose things like the node manager like this, as needed or to just expose the cluster itself.  Thoughts?